text/0056-tx-fee-map-digest.md

```diff
@@ -0,0 +1,95 @@
+- Feature Name: tx-fee-map-digest
+- Start Date: 2022-11-03
+- MCIP PR: https://github.com/mobilecoinfoundation/mobilecoin/pull/2793
+- Tracking Issue: https://github.com/mobilecoinfoundation/mobilecoin/issues/2611
+
+# Summary
+[summary]: #summary
+
+Modify the `Tx` struct to include a digest of the minimum fee map, which provides a mitigation against a possible information leak attack that can be executed by a malicious node operator.
+
+A security audit performed by Trail of Bits has identified a potential information leak attack: The consensus enclave fee map can be changed by the untrusted side, by restarting the node. If the attacker controls a node and is able to get the user to submit their Tx once with one fee map, and once with another, then the priority value leaks information about what the token id of the Tx is.
+
+In principle they could do this without the user knowing -- they would restart the node and it would not be able to attest to the peers after changing the fee map, but it could still accept Tx proposals and see the WellFormedTxContext which conatins the priority number. Then they turn the node back to normal. The user's app retries when their transaction doesn't land, and the same Tx comes back to their node, now with the regular fee map.
+
+This requires having root on a consensus node, but the threat model is that attackers with root on the consensus node should not be able to see the user's token id.
+
+We view this as a time-of-check-vs-time-of-use ([TOCTOU](https://en.wikipedia.org/wiki/Time-of-check_to_time-of-use)) bug, and we try to fix it a way that those bugs are sometimes fixed. The user, who checked the fee map, also makes an assertion about what the fee-map is when they try to use the system (by submitting a Tx). Before accepting the transaction, the enclave checks if the user's assertion is correct, and rejects the Tx without creating a WellFormedTxContext if the assertion is not correct. (This is sort of like dealing with races when manipulating atomic variables by using compare-and-swap.)
```

Comment:
:wrenche: 
* Not sure on this one, changed "fix it a way" -> "fix it in a way"
* "fee-map" to "fee map" it seems everywhere else it's 2 words
* Again not sure on the markup for "WellFormedTxContext"
```suggestion
We view this as a time-of-check-vs-time-of-use ([TOCTOU](https://en.wikipedia.org/wiki/Time-of-check_to_time-of-use)) bug, and we try to fix it in a way that those bugs are sometimes fixed. The user, who checked the fee map, also makes an assertion about what the fee map is when they try to use the system (by submitting a Tx). Before accepting the transaction, the enclave checks if the user's assertion is correct, and rejects the Tx without creating a `WellFormedTxContext` if the assertion is not correct. (This is sort of like dealing with races when manipulating atomic variables by using compare-and-swap.)
```